osu.Game/Beatmaps/Formats/OsuLegacyDecoder.cs

```diff
@@ -220,7 +222,7 @@ private void handleDifficulty(Beatmap beatmap, string line)
         /// <param name="line">The line which may contains variables.</param>
         private void decodeVariables(ref string line)
         {
-            while (line.IndexOf('$') >= 0)
+            if (line.IndexOf('$') >= 0)

```

Comment:
Was this resulting in an infinite loop? If so can you provide the file? I can imagine this might result in an infinite loop in the case of `$` not being at the start of the split strings right below this, however the while loop is intended because it allows nesting of variables, e.g.

```
a=5
b=$a+5
```
where `$b` should be decoded as
```
$b
=$a+5
=5+5
```
So instead of changing this to an `if`, it should instead be checking if the string was modified after each iteration.